sdk/core/azure-core/inc/azure/core/exception.hpp

```diff
@@ -145,10 +133,23 @@ namespace Azure { namespace Core {
      *
      */
     ~RequestFailedException() = default;
-
-  private:
-    std::string GetRawResponseField(
-        std::unique_ptr<Azure::Core::Http::RawResponse>& rawResponse,
-        std::string fieldName);
   };
+
+  namespace _internal {
+    class ExceptionFactory final {
```

Comment:
@JeffreyRichter what do you think about this name and approach rather than a public ctor?